apps/src/craft/code-connection/craft.js

```diff
@@ -119,13 +120,30 @@ Craft.runButtonClick = function () {
 Craft.executeUserCode = function () {
   let codeBlocks = Blockly.mainBlockSpace.getTopBlocks(true);
   const code = Blockly.Generator.blocksToCode('JavaScript', codeBlocks);
+  let interpreter;
 
   const evalApiMethods = {
     log: function (blockID, value) {
       studioApp().highlight(blockID);
       console.log('Logged: ' + value);
     },
+    prompt: function (blockID, callback) {
+      studioApp().highlight(blockID);
+      const value = prompt('Enter a value:');
+      setTimeout(() => {
+        callback(value);
+        interpreter.run();
+      }, 0);
+    },
+    delay: function (blockID, milliseconds, callback) {
+      studioApp().highlight(blockID);
+      setTimeout(() => {
+        callback();
+        interpreter.run();
+      }, milliseconds);
+    }
   };
 
-  codegen.evalWith(code, evalApiMethods);
+  codegen.asyncFunctionList = [evalApiMethods.prompt, evalApiMethods.delay];
```

Comment:
Native-is-async functions should be added here, so the interpreter knows to pass in the extra `callback` param.  Docs here: https://neil.fraser.name/software/JS-Interpreter/demos/async.html